eth_bridge/ic/src/eth_proxy/src/common/cap.rs

```diff
@@ -0,0 +1,35 @@
+use cap_sdk::{insert, IndefiniteEvent};
+
+use crate::proxy::{ToEvent, STATE};
+
+use super::types::ClaimableMessage;
+
+pub async fn insert_claimable_asset(message: ClaimableMessage) {
+    // add message to STATE claimable messages queue.
+    STATE.with(|s| {
+        s.add_claimable_message(message.clone());
+    });
+
+    // flush STATE claimable messages queue
+    register_messages().await
+}
+
+pub async fn register_messages() {
+    let mut pending_registrations = STATE.with(|s| s.get_all_claimable_messages());
```

Comment:
Then we can use the `insert_many_sync` which takes in a iterator of `pending_registrations`

Reply:
Refer to my comment about flushing below

https://github.com/Psychedelic/terabethia/pull/62#discussion_r975513452